shared_model/interfaces/iroha_internal/transaction_batch_factory.hpp

```diff
@@ -33,7 +33,8 @@ namespace shared_model {
        */
       template <typename TransactionValidator,
                 typename FieldValidator = validation::FieldValidator>
-      static iroha::expected::Result<TransactionBatch, std::string>
+      static iroha::expected::Result<std::unique_ptr<TransactionBatch>,
```

Comment:
What's the profit of class? Maybe just put functions in namespace?

Reply:
These methods should be virtual, so they are required to be in a class.

Reply:
but they aren't :(
would be nice to add a todo mark then